advanced/wallets/react-wallet-v2/src/utils/EIP155WalletUtil.ts

```diff
@@ -41,3 +43,48 @@ export function createOrRestoreEIP155Wallet() {
     eip155Addresses
   }
 }
+
+/**
+ * Get wallet for the address in params
+ */
+export const getWallet = async (params: any) => {
+  const eoaWallet = eip155Wallets[getWalletAddressFromParams(eip155Addresses, params)]
+  if (eoaWallet) {
+    return eoaWallet
+  }
+
+  /**
+   * Smart accounts
+   */
+  const chainId = params?.chainId?.split(':')[1]
+  console.log('Chain ID', { chainId })
+  console.log('PARAMS', { params })
+
+  const address = getWalletAddressFromParams(
+    Object.keys(smartAccountWallets)
+      .filter(key => {
+        const parts = key.split(':')
+        return parts[0] === chainId
+      })
+      .map(key => {
+        return key.split(':')[1]
+      }),
+    params
+  )
+  if (!address) {
+    console.log('Library not initialized for requested address', {
+      address,
+      values: Object.keys(smartAccountWallets)
+    })
+    throw new Error('Library not initialized for requested address')
+  }
+  const lib = smartAccountWallets[`${chainId}:${address}`]
+  if (lib) {
+    return lib
+  }
+  console.log('Library not found', {
+    target: `${chainId}:address`,
+    values: Object.keys(smartAccountWallets)
+  })
+  throw new Error('Cannot find wallet for requested address')
+}
```

Comment:
This code is copy pasted from `EIP155RequestHandlerUtil.ts`. If we extract it to different place, lets make sure to  use `getWallet` reference in the initial implementation

Reply:
sure, will be change it in the initial reference implementation to use this.